#!/usr/bin/env python

from dynamic_reconfigure.parameter_generator_catkin import *

PACKAGE_NAME = "squirrel_navigation"

gen = ParameterGenerator()
gen.add("max_linear_velocity", double_t, 0, "Maximum linear velocity allowed", 0.5, 0.0, 10.0)
gen.add("max_angular_velocity", double_t, 0, "Maximum angular velocity allowed", 0.7, 0.0, 10.0)
gen.add("linear_smoother", double_t, 0, "Parameter for linear smoothing of the trajectory", 0.95, 0.0, 1.0)
gen.add("angular_smoother", double_t, 0, "Parameter for angular smoothing of the trajectory", 0.95, 0.0, 1.0)
gen.add("time_scaler", double_t, 0, "Rescaler for interpolation time", 1.25, 1.0, 10.0)
gen.add("waypoints_heading_lookahead", int_t, 0, "Lookahead for updates", 5, 0, 100)

exit(gen.generate(PACKAGE_NAME, "squirrel_navigation", "MotionPlanner"))